cdap-gateway/src/test/java/co/cask/cdap/gateway/handlers/metrics/MetricsHandlerTestRun.java

```diff
@@ -541,6 +541,24 @@ public void testInterpolate() throws Exception {
     metricStore.delete(deleteQuery);
   }
 
+  @Test
+  public void testResolutionInResponse() throws Exception {
+    long start = 1;
+
+    String url = "/v3/metrics/query?" + getTags("resolutions", "WordCount1", "WordCounter", "splitter") +
+      "&metric=system.reads&resolution=auto&start=" + (start - 1) + "&end="
+      + (start + 36000);
+    MetricQueryResult queryResult = post(url, MetricQueryResult.class);
+    Assert.assertEquals(3600, queryResult.getResolution());
```

Comment:
also add a test case, where you skip resolution. then check for resolution in batched request as well. you can edit the existing test case if you may for that.